### PR TITLE
Add failure message to PR if after merge build fails

### PR DESCRIPTION
**Pull request by mwperina**
Adds a failure message to PR as a new comment, if packit build fails
after merge of the PR.

Signed-off-by: Martin Perina <mperina@redhat.com>

### CODE REVIEW

File: .packit.yml
Comment (engelmi) [inline .packit.yml:77]: If I understand the `trigger: commit` correctly, this comment will be added each time the build(s) fail - also for not yet merged PRs, right? I think it would be great to have the comment only when the PR is not yet merged. 
I'd suggest to duplicate the `copr_build` job - one without the notification and one with. To filter based on merged to main then add `branch: '(?!main)'` to the former and `branch: main` to the latter (with the notification). 
What do you think?

File: .packit.yml
Comment (engelmi) [inline .packit.yml:78]: What do you think about tagging people or a group, e.g. `@eclipse-bluechi/automotive-bluechi-project-leads`?

File: .packit.yml
Comment (mwperina) [reply .packit.yml:77]: If the PR is not merged, then you can see errors *Checks* section at the bottom of the PR, so it doesn't seem to me worth to add additional comment about error.

But when PR is merged (that's `trigger:commit` meaning), then the only way to find a failure is to take a look at https://copr.fedorainfracloud.org/coprs/g/centos-automotive-sig/bluechi-snapshot/builds/ manually and check. So adding a comment to PR about failure after merge would make the error much more visible.

File: .packit.yml
Comment (mwperina) [reply .packit.yml:78]: Do we have such group?

File: .packit.yml
Comment (engelmi) [reply .packit.yml:77]: Based on 
https://packit.dev/docs/configuration/upstream/upstream_koji_build#supported-triggers
`trigger: commit` will lead to the job being run on each commit for the specific branch. So if no branch is specified, its basically for all. 

> If the PR is not merged, then you can see errors Checks section at the bottom of the PR, so it doesn't seem to me worth to add additional comment about error.

I agree. Thats why I would distinguish the commit based on branch - commit to main means a PR is merged.

File: .packit.yml
Comment (engelmi) [reply .packit.yml:78]: Yes, there is the project leads group (that I linked) and one for committer. I think committer in general includes too many people and leads (only 3 people) not enough. Either we ask one to be created or we simply list the volunteers directly.

File: .packit.yml
Comment (mwperina) [reply .packit.yml:77]: Job
```
  - job: copr_build
    trigger: pull_request
```
is executed when PR is created or updated, but it's not executed when PR is merged.

Job
```
  - job: copr_build
    trigger: commit
```
is executed only when a commit is merged to the https://github.com/eclipse-bluechi/bluechi repo, which happens only when PR is merged. So why should we we distinguish branches here? We want to know
even if we for example merge commit to stable branch like `bluechi-0.6.z` and build fails, right?
Or am I missing something?

File: .packit.yml
Comment (mwperina) [reply .packit.yml:78]: Hmm, but just adding a comment to the PR GH should send an email to all CODEOWNERS (as they are assigned to PR as reviewers) and all other engineers who posted a comment on a PR.
But if you think it's worth to tag project leads group directly, I can add it to the message

Comment (engelmi) [reply .packit.yml:78]: Yes, you are right. I don't receive emails for BlueChi, but other repos, interestingly. The GH notifications on the website are definitely there then. We can still improve later if necessary.

File: .packit.yml
Comment (engelmi) [reply .packit.yml:77]: From https://packit.dev/docs/configuration/upstream/copr_build#supported-triggers:
```
commit - reacts to new commits to the specified branch
```
and further down in https://packit.dev/docs/configuration/upstream/copr_build#optional-parameters
```
branch - the name of the branch we want to build for when using commit trigger (defaults to the repository's default branch) 
```

> We want to know even if we for example merge commit to stable branch like bluechi-0.6.z and build fails, right?

Yes. But currently this `copr_build` for commits does not apply for other stable branches such as `bluechi-0.6.z`, right?

> So why should we we distinguish branches here? 

I was under the assumption that `trigger: commits` applies to all branches when not specifying one. But according to the explanation above, it applies only for `main` at the moment. Not sure how this works with forks. So maybe adding `branch` for main and other stable branches would be what we want?

Comment (mwperina) [reply .packit.yml:77]: If I understand it correctly copr builds on specific stable branch when we create this branch and it includes packit yaml configuration file.
Regarding forks, they are different repositories, so even though you push a change to them, this trigger is on executed.
I've looked at projects mentioned at https://packit.dev/docs/configuration/examples#get-inspired and there are differences: some projects don't have a branch specified, some have configuration for both development and stable branch included in the development branch and some have configuration for development branch in development branch and for stable branch in stable branch (this approach is probably most understandable).

So I will specify main branch for packit configuration in main and when we create a stable branch, we will overwrite this option in packit configuration for this branch

**Review by engelmi — APPROVED** [review]
LGTM

**Pull request merged**
